pkg/install/stack/common/storageclass.go

```diff
@@ -31,95 +33,175 @@ import (
 			classes := storagev1.StorageClassList{}
 			if err := kubeClient.List(ctx, &classes); err != nil {
-				return *s, fmt.Errorf("cannot list existing StorageClasses: %w", err)
+				return fmt.Errorf("cannot list existing StorageClasses: %w", err)
 			}
 
 			for _, class := range classes.Items {
 				if isDefaultStorageClass(class) {
 					logger.Infof("Duplicating existing default class %q…", class.Name)
 
-					s = class.DeepCopy()
-					s.Annotations = map[string]string{}
-					s.Labels = map[string]string{}
-					s.ResourceVersion = ""
-					s.Name = name
+					sc.Annotations = class.Annotations
+					sc.Labels = class.Labels
+					sc.Provisioner = class.Provisioner
+					sc.Parameters = class.Parameters
+					sc.ReclaimPolicy = class.ReclaimPolicy
+					sc.MountOptions = class.MountOptions
+					sc.AllowVolumeExpansion = class.AllowVolumeExpansion
+					sc.VolumeBindingMode = class.VolumeBindingMode
+					sc.AllowedTopologies = class.AllowedTopologies
 
-					return *s, nil
+					delete(sc.Annotations, "storageclass.kubernetes.io/is-default-class")
+					delete(sc.Annotations, "storageclass.beta.kubernetes.io/is-default-class")
+
+					return nil
 				}
 			}
 
-			return *s, errors.New("cannot duplicate existing default StorageClass, because none of the classes are marked as the default")
+			return errors.New("cannot duplicate existing default StorageClass, because none of the classes are marked as the default")
 		},
-		"gce": func(_ context.Context, _ *logrus.Entry, _ ctrlruntimeclient.Client, name string) (storagev1.StorageClass, error) {
-			s := storagev1.StorageClass{
-				Parameters: map[string]string{},
+		kubermaticv1.GCPCloudProvider: func(_ context.Context, _ *logrus.Entry, _ ctrlruntimeclient.Client, sc *storagev1.StorageClass, csiDriverName string) error {
+			if csiDriverName == "" { // = in-tree CSI
+				sc.Provisioner = "kubernetes.io/gce-pd"
+			} else { // out-of-tree CSI
+				sc.Provisioner = csiDriverName
 			}
-			s.Name = name
-			s.Provisioner = "kubernetes.io/gce-pd"
-			s.Parameters["type"] = "pd-ssd"
 
-			return s, nil
+			sc.Parameters["type"] = "pd-ssd"
+			sc.VolumeBindingMode = &waitForFirstCustomer
+
+			return nil
 		},
-		"aws": func(_ context.Context, _ *logrus.Entry, _ ctrlruntimeclient.Client, name string) (storagev1.StorageClass, error) {
-			s := storagev1.StorageClass{
-				Parameters: map[string]string{},
+		kubermaticv1.AWSCloudProvider: func(_ context.Context, _ *logrus.Entry, _ ctrlruntimeclient.Client, sc *storagev1.StorageClass, csiDriverName string) error {
+			if csiDriverName == "" { // = in-tree CSI
+				sc.Provisioner = "kubernetes.io/aws-ebs"
+			} else { // out-of-tree CSI
+				sc.Provisioner = csiDriverName
 			}
-			s.Name = name
-			s.Provisioner = "kubernetes.io/aws-ebs"
-			s.Parameters["type"] = "gp2"
 
-			return s, nil
+			sc.Parameters["type"] = "io1"
```

Comment:
also needs iopsPerGB, check out what we have on the latest snapshot (20 is the value we went with).